tools/drdeploy.c

```diff
@@ -1270,6 +1282,20 @@ _tmain(int argc, TCHAR *targv[])
     /* we re-read the tool list if the root, platform or toolconfig dir change */
     read_tool_list(dr_toolconfig_dir, dr_platform);
 
+#if defined(LINUX) && !defined(ANDROID) && (defined(DRCONFIG) || defined(DRRUN))
+    /* XXX i#5431: Workaround for an rseq issue with glibc 2.35 which makes many
+     * apps fail up front.  We expect to remove this once the real fix is in place.
+     */
+#    include <gnu/libc-version.h>
+    const char *libc_ver = gnu_get_libc_version();
+    if (libc_ver[0] != '\0' && libc_ver[0] >= '2' && libc_ver[1] == '.' &&
```

Comment:
Didn't pay attention to this before: this condition won't work for all glibc versions >= 2.35, e.g. it would be false for 2.41.

Reply:
I hope this is all ripped out and replaced with a long-term solution long before that.

Reply:
Fair enough.